src/pages/Profile/index.js

```diff
@@ -23,90 +37,336 @@ import avatar from '../../assets/avatar.svg';
 class Profile extends React.Component {
   constructor(props) {
     super(props);
+    this.ref = {
+      name: React.createRef(),
+      email: React.createRef(),
+      position: React.createRef(),
+    };
     this.state = {
-      isChecked: true,
+      editView: profileData.editView,
+      displayName: profileData.displayName,
+      emailVal: profileData.emailVal,
+      positionVal: profileData.positionVal,
+      accountStatus: profileData.accountStatus,
+      companyID: profileData.companyID,
+      color: '#0080005e',
+      isModalOpen: false,
+    };
+
+    this.accountStatus = [
+      { value: 'Active', label: 'Active', disabled: false },
+      { value: 'Inactive', label: 'Inactive', disabled: false },
+    ];
+
+    this.handleModalToggle = () => {
+      this.setState(({ isModalOpen }) => ({
+        isModalOpen: !isModalOpen,
+      }));
+    };
+
+    this.onaccountStatusChange = accountStatus => {
+      this.setState({
+        accountStatus,
+        color: accountStatus === 'Active' ? '#0080005e' : '#ff000052',
+      });
     };
-  }
 
-  componentDidMount = () => {};
+    this.edit = () => {
+      const { editView } = this.state;
+      this.setState({
+        editView: !editView,
+      });
+    };
+
+    this.saveEdit = () => {
+      const { editView, displayName, emailVal, positionVal } = this.state;
+      const { name, email, position } = this.ref;
+      const editedName = name.current.value ? name.current.value : displayName;
+      const editedEmail = email.current.value ? email.current.value : emailVal;
+      const editedPosition = position.current.value ? position.current.value : positionVal;
 
-  handleChange = isChecked => {
-    this.setState({ isChecked });
-  };
+      this.setState({
+        editView: !editView,
+        displayName: editedName,
+        emailVal: editedEmail,
+        positionVal: editedPosition,
+      });
+    };
+
+    this.cancleEdit = () => {
+      const { editView } = this.state;
+      this.setState({
+        editView: !editView,
+      });
+    };
+
+    this.removeTag = tag => {
```

Comment:
Can we consider removing `this.removeTag` as it is not being referenced anywhere else?

Reply:
removed